### PR TITLE
[Quorum Store] A couple metrics to observe backpressure better

### PR DESCRIPTION
**Pull request by bchocho**
### Description

A couple metrics to observe backpressure better

### CODE REVIEW

File: consensus/src/quorum_store/batch_generator.rs [hunk 1]
```diff
@@ -250,6 +251,9 @@ impl BatchGenerator {
         } else {
             counters::PULLED_TXNS_COUNT.inc();
             counters::PULLED_TXNS_NUM.observe(pulled_txns.len() as f64);
+            if pulled_txns.len() as u64 == max_count {
+                counters::BATCH_PULL_FULL_TXNS.observe(max_count as f64)
```
Comment (ibalajiarun) [inline consensus/src/quorum_store/batch_generator.rs:255]: why not set this check as a boolean label to `counters::PULLED_TXNS_COUNT`? is max_count not available else where?

**Review by ibalajiarun — APPROVED** [review]

File: consensus/src/quorum_store/batch_generator.rs (hunk 1 above)
Comment (bchocho) [reply consensus/src/quorum_store/batch_generator.rs:255]: Yeah, this is actually the first time putting max_count in metrics.

**Review by danielxiangzl — APPROVED** [review]
LGTM

File: .github/linters/semgrep/pull-request-target-code-checkout.yaml [hunk 2]
```diff
@@ -47,6 +47,15 @@ rules:
             ...
             $JOBNAME:
               ...
+      - pattern-not-inside: |
```
Comment (JoshLind) [inline .github/linters/semgrep/pull-request-target-code-checkout.yaml:50]: FYI: you should probable rebase 😄

Comment (JoshLind) [reply .github/linters/semgrep/pull-request-target-code-checkout.yaml:50]: But, you already have two stamps 😄

Comment (bchocho) [reply .github/linters/semgrep/pull-request-target-code-checkout.yaml:50]: I'll rebase and land :)

**Review by JoshLind — APPROVED** [review]

**Pull request merged**